pkg/discovery/worker/aggregation/container_utilization_data_aggregator.go

```diff
@@ -0,0 +1,92 @@
+package aggregation
+
+import (
+	"fmt"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+	"math"
+)
+
+var (
+	maxUtilizationDataStrategy = "maxUtilizationData"
+	allUtilizationDataStrategy = "allUtilizationData"
+
+	DefaultContainerUtilizationDataAggStrategy = maxUtilizationDataStrategy
+
+	// Map from the configured utilization data aggregation strategy to utilization data aggregator
+	ContainerUtilizationDataAggregators = map[string]ContainerUtilizationDataAggregator{
+		maxUtilizationDataStrategy: &maxUtilizationDataAggregator{aggregationStrategy: "max utilization data strategy"},
+		allUtilizationDataStrategy: &allUtilizationDataAggregator{aggregationStrategy: "all utilization data strategy"},
+	}
+)
+
+// ContainerUtilizationDataAggregator interface represents a type of container utilization data aggregator
+type ContainerUtilizationDataAggregator interface {
+	// AggregationStrategy returns aggregation strategy of this data aggregator
+	AggregationStrategy() string
+	// Aggregate aggregates commodities utilization data based on the given list of commodity DTOs of a commodity type
+	// and aggregation strategy, and returns aggregated utilization data which contains utilization data points, last
+	// point timestamp milliseconds and interval milliseconds
+	Aggregate(commodities []*proto.CommodityDTO, lastPointTimestampMs int64) ([]float64, int64, int32, error)
```

Comment:
I dont see the 3rd argument used in the code.

Reply:
The 3rd returned argument means the interval milliseconds of the data points we collect every discovery cycle. For example, we collect metrics data points every 1mins for Cloud probe. For kubeturbo for now, we collect only one set of data points every discovery cycle so the 3rd returned value is always 0. I reserved the 3rd argument here in case we want to collect multiple metrics data points within a discovery cycle in the future.

I think I can only return the first argument `[]float64` in this function to avoid confusion. If we need the interval milliseconds in the future, we can set it to wherever properly.